packages/@divvi/mobile/src/swap/UnfavorableRateBottomSheet.tsx

```diff
@@ -38,21 +43,112 @@ export default function UnfavorableRateBottomSheet({
           type={BtnTypes.PRIMARY}
           size={BtnSizes.FULL}
           text={t('swapUnfavorableRateBottomSheet.cancel')}
+          touchableStyle={{ height: BUTTON_HEIGHT }}
         />
-        <Button
-          onPress={() => {
+        <SlideButton
+          onComplete={() => {
             forwardedRef.current?.close()
             onConfirm()
           }}
-          type={BtnTypes.SECONDARY}
-          size={BtnSizes.FULL}
-          text={t('swapUnfavorableRateBottomSheet.confirm')}
         />
       </View>
     </BottomSheet>
   )
 }
 
+const SlideButton = ({ onComplete }: { onComplete: () => void }) => {
+  const { t } = useTranslation()
+  const slideThreshold = useRef(1)
+  const pan = useRef(new Animated.Value(0)).current
+  const [completed, setCompleted] = useState(false)
+
+  const onCompleteRef = useRef(onComplete)
+
+  // ensure the latest onComplete is used by the pan responder
+  useEffect(() => {
+    onCompleteRef.current = onComplete
+  }, [onComplete])
+
+  const panResponder = useRef(
+    PanResponder.create({
+      onMoveShouldSetPanResponder: () => !completed,
+      onPanResponderMove: Animated.event([null, { dx: pan }], { useNativeDriver: false }),
```

Comment:
I think we should be able to refactor the logic a bit and `useNativeDriver: true` since we're only animating `transform` which works with the native driver.
This way we'll have better performance even on lower end devices.

Reply:
I've adjusted timing / spring to use the native driver. It seems like `onPanResponderMove` doesn't work with native driver (https://github.com/facebook/react-native/issues/13377#issuecomment-292657541). Seems like the suggestion is to use another gesture tracking package, maybe like react-native-gesture-handler. Is this approach significantly bad on lower end devices? If not, I am thinking of leaving it as is, given we didn't want to spend too much time on this.

Reply:
Sounds good, yes hard to tell without trying on a lower end device.
We can improve later if it's a noticeable problem 👍